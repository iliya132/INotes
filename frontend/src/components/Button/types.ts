import { MouseEventHandler } from "react";

export interface IButtonProps {
    title: string,
    className?: string,
    disabled?: boolean;
    onClick?: MouseEventHandler<HTMLButtonElement>;
}