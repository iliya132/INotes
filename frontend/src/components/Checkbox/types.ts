/* eslint-disable no-unused-vars */
export interface ICheckboxProps {
    defaultValue?: boolean;
    id: string;
    className?: string;
    onChanged? : (newValue:boolean) => void;
    title?: string;
}
