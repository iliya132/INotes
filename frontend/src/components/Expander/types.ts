export interface IExpanderProps {
    isExpanded: boolean;
    expanderBody: JSX.Element;
    children: JSX.Element;
    onClick?: () => void;
}
