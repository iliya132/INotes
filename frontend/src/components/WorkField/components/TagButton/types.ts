import { INote } from "../../../../store/types";

export default interface ITagButtonProps {
    note: INote | null | undefined;
}

export interface ITag{
    label: string,
    value: string
}