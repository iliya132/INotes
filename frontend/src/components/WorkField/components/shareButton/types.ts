import { INote } from "../../../../store/types";

export default interface IShareBtnProps {
    note: INote | null | undefined;
}
