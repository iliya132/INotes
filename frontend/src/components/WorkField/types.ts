/* eslint-disable no-unused-vars */
import { INote } from "../../store/types";

export interface IWorkfieldProps {
    note?: INote;
}

export enum WfAction {
    Bold,
    Italic,
    Underscoped,
    Dotted,
    Number,
    H1,
    H2,
    H3,
    Code,
    Table
}
