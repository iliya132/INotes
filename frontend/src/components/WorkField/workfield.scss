@import '../../Misc/colors.scss';

.work-field {
    display: grid;
    box-sizing: border-box;
    grid-template-rows: 50px 1fr;
    width: 100%;
    min-height: 100px;
    max-height: 100%;
    border-radius: 9px;
}

.workfield-actions {
    display: flex;
    justify-content: space-between;
    font-size: 18px;
    font-weight: 500;
}

.workfield-actions-container {
    display: flex;
    margin-right: 5px;
    margin-left: auto;
}

.workfield-buttons {
    display: flex;
    margin-top: 6px;
    margin-left: auto;
}

.stash-changes-btn {
    display: none; //TODO
    margin-right: 11px;
    border-radius: 6px;
    background-color: $Gray;
    color: $TextColor;
    text-transform: none;

    &:hover {
        background-color: #bebcb5;
    }
}

.apply-changes-btn {
    margin: 5px;
    border-radius: 6px;
    text-transform: none;

    &:hover {
        background-color: #2a3f35;
    }
}

.workfield-content {
    display: grid;
    grid-template-columns: 1fr 1fr;
}

.workfield-content-read-mode {
    display: grid;
    grid-template-columns: 1fr;
}

.workfield-content-edit-field {
    display: grid;
    grid-template-rows: 1fr;
    border-right: 1px solid $Gray;
}

.workfield-content-edit-field-actions {
    display: flex;
    margin-left: 5px;
}

.wf-action-btn {
    margin-right: 3px;
}

.workfield-content-display-field {
    padding: 15px;
    background-color: white;
}

.workfield-content-edit-field-textarea {
    margin: 5px 5px 0;

    textarea {
        box-sizing: border-box;
        width: 100%;
        height: 98%;
        margin: auto;
        padding: 15px;
        border: none;
        border-radius: 5px;
        outline: none;
        resize: none;

        &::-webkit-scrollbar {
            width: 5px;
            cursor: all-scroll;
        }

        &::-webkit-scrollbar-track {
            background: white;
        }

        &::-webkit-scrollbar-thumb {
            border-radius: 20px;
            background-color: $G4;
        }
    }
}

.note-name-edit-field {
    width: 100%;
    height: 30px;
    border: none;
    border-bottom: none;
    outline: none;
    background-color: transparent;
    color: $TextColor;
    font-size: 13px;
    font-size: 20px;
    line-height: 18px;
    text-align: left;
}

.save-btn {
    background-color: $Background-light;
}

.render-mode {
    width: 60%;
    margin-right: auto !important;
    margin-left: auto !important;
}

.reader-mode-invisible {
    display: none;
}

.markdown-help-container {
    min-width: 450px;
    padding: 15px;
    border: 1px solid $BlackCommon;
    background-color: white;
    text-align: left;

    & ul {
        margin: 0;
        padding: 0;
        list-style: none;

        & li {
            display: flex;
            flex-direction: row;
            justify-content: space-between;

            & .markdown-help-container-example {
                display: inline-block;
                margin-left: auto;
                padding: 3px;
                border: 1px solid gray;
                border-radius: 5px;
                background-color: $Gray;
            }
        }
    }
}
