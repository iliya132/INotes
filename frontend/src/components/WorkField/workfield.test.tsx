/* eslint-disable no-undef */
import React from 'react';
import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Workfield } from './workfield';
import { INote, INotebook, INoteDTO } from '../../store/types';
import notesController from '../../controllers/NotesController';
import filesController from '../../controllers/FilesController';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { authSlice } from '../../store/reducers/authReducer';
import { notesSlice } from '../../store/reducers/notebooksReducer';
import { RootState } from '../../store/store';

jest.mock('../../controllers/NotesController', () => ({
    removeNote: jest.fn(),
    updateNote: jest.fn(),
}));

jest.mock('../../controllers/FilesController', () => ({
    uploadfile: jest.fn()
}))

let initialState: RootState = {
    authReducer: {
        error: '',
        isAuth: true,
        user: { roles: ['user'], userName: 'test-user', avatarUrl: '/default.png' },
        validation: {
            isSucceded: true,
            errors: {},
        },
    },
    notebooksReducer: {
        notebooks: [],
        allNotes: [],
        userTags: [],
        selectedTags: []
    },
};

let store = configureStore({
    reducer: {
        authReducer: authSlice(initialState.authReducer).reducer,
        notebooksReducer: notesSlice(initialState.notebooksReducer).reducer,
    },
});

const notebook: INotebook = {
    color: 'red',
    id: 1,
    name: 'test-notebook',
};
let note: INote = {
    content: 'test-content',
    id: 0,
    isNew: false,
    isPublicUrlShared: false,
    name: 'test-name',
    parent: notebook,
    publicUrl: 'test-public-url',
};

describe('workfield tests', () => {
    test('can render', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const workfield = document.body.querySelector('.work-field');
        expect(workfield).not.toBeNull();
    });

    test('match snapshot', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const workfield = document.body.querySelector('.work-field');
        expect(workfield).not.toBeNull();
        expect(workfield?.outerHTML).toMatchSnapshot('workfield snapshot');
    });

    test('can be rendered when note is null', () => {
        render(<Provider store={store}><Workfield note={null} /></Provider>);
        const workfield = document.body.querySelector('.work-field');
        expect(workfield).not.toBeNull();
    });

    test('can be rendered when note is undefined', () => {
        render(<Provider store={store}><Workfield note={undefined} /></Provider>);
        const workfield = document.body.querySelector('.work-field');
        expect(workfield).not.toBeNull();
    });

    test('helpers bold', async () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('bold-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('**test-content**');
    });

    test('helpers italic', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('italic-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('*test-content*');
    });

    test('helpers underscoped', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('underscoped-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('_test-content_');
    });

    test('helpers h1', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('h1-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('# test-content');
    });

    test('helpers h2', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('h2-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('## test-content');
    });

    test('helpers h3', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('h3-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('### test-content');
    });

    test('helpers dotted list', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('dotted-list-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('* test-content');
    });

    test('helpers number list', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('list-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('1. test-content');
    });

    test('helpers table', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('table-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('| column1 | column2 |\n|-----|-----|\n| value | value |\ntest-content');
    });

    test('helpers code', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('code-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field');
        (input as HTMLTextAreaElement).select();

        fireEvent.click(btn);

        expect(input.textContent).toEqual('```\ntest-content\n```');
    });

    test('helpers markdown', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('markup-icon');
        expect(btn).not.toBeNull();

        fireEvent.click(btn);

        const popup = document.querySelector('[data-testid=markdown-help]');
        expect(popup).not.toBeNull();

        expect(popup!.outerHTML).toMatchSnapshot('markdown-help-snapshot');
    });

    test('can delete note', () => {
        let isRemoved = false;
        // eslint-disable-next-line no-unused-vars
        const remove = jest.fn((noteId: number) => {
            isRemoved = true;
        });
        notesController.removeNote = remove;
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('remove');
        expect(btn).not.toBeNull();

        fireEvent.click(btn);

        expect(isRemoved).toBe(true);
    });

    test('can copy note', () => {
        //TODO not implemented yet
        expect(true).toBe(true);
    });

    test('can tag note', () => {
        //TODO not implemented yet
        expect(true).toBe(true);
    });

    test('can go in reader mode', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('read-icon');
        expect(btn).not.toBeNull();

        fireEvent.click(btn);

        const workfield = document.querySelector('.workfield-content-read-mode');
        expect(workfield).not.toBeNull();

        const editArea = document.querySelector('.reader-mode-invisible');
        expect(editArea).not.toBeNull();

        expect(workfield).toMatchSnapshot('workfield-read-mode');
    });

    test('can save note', () => {
        let isUpdated = false;
        // eslint-disable-next-line no-unused-vars
        const save = jest.fn((_note: INoteDTO) => {
            isUpdated = true;
        });
        notesController.updateNote = save;
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('save-icon');
        expect(btn).not.toBeNull();

        fireEvent.click(btn);

        expect(isUpdated).toBe(true);
    });

    test('can fire onChange', () => {
        let isChanged = false;
        render(<Provider store={store}><Workfield note={note} onChange={() => (isChanged = true)} /></Provider>);

        const input = screen.getByTestId('input-field') as HTMLTextAreaElement;
        expect(input).not.toBeNull();
        fireEvent.change(input, { target: { value: 'test-content2' } });

        expect(input.value).toBe('test-content2');
        expect(isChanged).toBe(true);
    });

    test('can fire onSave', () => {
        let isUpdated = false;
        // eslint-disable-next-line no-unused-vars
        const save = jest.fn((_note: INoteDTO) => {
            isUpdated = true;
        });
        notesController.updateNote = save;
        let isOnSaved = false;
        render(<Provider store={store}><Workfield note={note} onSave={() => (isOnSaved = true)} /></Provider>);
        const btn = screen.getByTestId('save-icon');
        expect(btn).not.toBeNull();

        fireEvent.click(btn);

        expect(isUpdated).toBe(true);
        expect(isOnSaved).toBe(true);
    });

    test('when save without title then ok', () => {
        let isUpdated = false;
        let title = '';
        // eslint-disable-next-line no-unused-vars
        const save = jest.fn((_note: INoteDTO) => {
            isUpdated = true;
            title = _note.name;
        });
        notesController.updateNote = save;
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const btn = screen.getByTestId('save-icon');
        expect(btn).not.toBeNull();

        const input = screen.getByTestId('input-field') as HTMLTextAreaElement;
        expect(input).not.toBeNull();
        fireEvent.change(input, { target: { value: '' } });
        fireEvent.click(btn);

        expect(isUpdated).toBe(true);
        expect(title).toEqual('Без названия');
    });

    test('it renders on input', () => {
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const input = screen.getByTestId('input-field') as HTMLTextAreaElement;
        const renderedView = screen.getByTestId('pretty-view');
        expect(input.value).toEqual('test-content');

        fireEvent.change(input, { target: { value: '**test-bold**' } });
        expect(renderedView.innerHTML).toEqual('<p><strong>test-bold</strong></p>\n');

        fireEvent.change(input, { target: { value: '*test-italic*' } });
        expect(renderedView.innerHTML).toEqual('<p><em>test-italic</em></p>\n');

        fireEvent.change(input, { target: { value: '_test-underscoped_' } });
        expect(renderedView.innerHTML).toEqual('<p><u>test-underscoped</u></p>\n');

        fireEvent.change(input, { target: { value: '* test-dotted' } });
        expect(renderedView.innerHTML).toEqual('<ul>\n<li>test-dotted</li>\n</ul>\n');

        fireEvent.change(input, { target: { value: '1. test-number' } });
        expect(renderedView.innerHTML).toEqual('<ol>\n<li>test-number</li>\n</ol>\n');

        fireEvent.change(input, { target: { value: '# test-header' } });
        expect(renderedView.innerHTML).toEqual('<h1>test-header</h1>\n');

        fireEvent.change(input, { target: { value: '## test-header' } });
        expect(renderedView.innerHTML).toEqual('<h2>test-header</h2>\n');

        fireEvent.change(input, { target: { value: '### test-header' } });
        expect(renderedView.innerHTML).toEqual('<h3>test-header</h3>\n');

        fireEvent.change(input, { target: { value: '```\ntest-code\n```' } });
        expect(renderedView.innerHTML).toEqual('<pre class="hljs"><code>test-code\n</code></pre>\n');

        const tableInput = '| column1 | column2 |\n' + '|-----|-----|\n' + '| value | value |';
        const tableExpected =
            '<table>\n' +
            '<thead>\n' +
            '<tr>\n' +
            '<th>column1</th>\n' +
            '<th>column2</th>\n' +
            '</tr>\n' +
            '</thead>\n' +
            '<tbody>\n' +
            '<tr>\n' +
            '<td>value</td>\n' +
            '<td>value</td>\n' +
            '</tr>\n' +
            '</tbody>\n' +
            '</table>\n';
        fireEvent.change(input, { target: { value: tableInput } });
        expect(renderedView.innerHTML).toEqual(tableExpected);

        fireEvent.change(input, { target: { value: 'test-row1\ntest-row2' } });
        input.select();
        fireEvent.keyDown(input, { key: 'Tab', code: 'Tab', charCode: 9 });
        expect(input.value).toEqual("    test-row1\n    test-row2");

        fireEvent.keyDown(input, { key: 'Tab', code: 'Tab', charCode: 9, shiftKey: true });
        expect(input.value).toEqual("test-row1\ntest-row2");
    });

    test('can drop file', async () => {
        const blob: BlobPart[] = [new ArrayBuffer(10)]
        const file = new File(blob, 'test-filename')
        const dti = new MockedDataTransferItem(file)
        render(<Provider store={store}><Workfield note={note} /></Provider>);
        const input = screen.getByTestId('input-field') as HTMLTextAreaElement;

        const uploadMock = jest.fn((_files: File[], _noteId: number) => {
            return Promise.resolve({ fileName: "linkToDownload" } as any)
        })
        fireEvent.select(input);

        filesController.uploadfile = uploadMock

        await act(async () => {
            fireEvent.drop(input, { dataTransfer: { items: [dti] } })
            await waitFor(() => expect(filesController.uploadfile).toHaveBeenCalledTimes(1))
        })

        expect(input.value).toContain('[fileName](linkToDownload)')
    })
});

class MockedDataTransferItem {
    private file: File;

    constructor(file: File) {
        this.file = file;
    }

    public getAsFile(): File | undefined {
        return this.file;
    }
}
