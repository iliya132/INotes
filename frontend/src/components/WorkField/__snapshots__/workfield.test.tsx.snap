// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workfield tests can go in reader mode: workfield-read-mode 1`] = `
<div
  class="workfield-content-read-mode"
>
  <div
    class="workfield-content-edit-field reader-mode-invisible"
  >
    <div
      class="workfield-content-edit-field-textarea"
    >
      <textarea
        data-testid="input-field"
      >
        test-content
      </textarea>
    </div>
  </div>
  <div
    class="workfield-content-display-field render-mode"
    data-testid="pretty-view"
  >
    <p>
      test-content
    </p>
    

  </div>
</div>
`;

exports[`workfield tests helpers markdown: markdown-help-snapshot 1`] = `"<div class=\\"markdown-help-container\\" data-testid=\\"markdown-help\\"><ul><li>Жирный текст <div class=\\"markdown-help-container-example\\">**Text**</div></li><hr><li>Курсивный текст <div class=\\"markdown-help-container-example\\">*Text*</div></li><hr><li>Подчеркнутый текст <div class=\\"markdown-help-container-example\\">_Text_</div></li><hr><li>Заголовок 1 уровня <div class=\\"markdown-help-container-example\\"># Text</div></li><hr><li>Заголовок 2 уровня <div class=\\"markdown-help-container-example\\">## Text</div></li><hr><li>Заголовок 3 уровня <div class=\\"markdown-help-container-example\\">### Text</div></li><hr><li>Список с точками <div class=\\"markdown-help-container-example\\">* Text</div></li><hr><li>Список с цифрами <div class=\\"markdown-help-container-example\\">1. Text</div></li><hr><li>Блок кода <div class=\\"markdown-help-container-example\\">\`\`\` javascript <br> Text <br>\`\`\`</div></li><hr><li>Однострочный код <div class=\\"markdown-help-container-example\\">\`Text\`</div></li><hr><li>Таблица <div class=\\"markdown-help-container-example\\">| header1 | header2 | <br> |-----|-----| <br>| value1 | value2 |</div></li><hr><li>Ссылка <div class=\\"markdown-help-container-example\\">[shown text](http://link_here.com)</div></li><hr><li>Горизонтальная линия <div class=\\"markdown-help-container-example\\">---</div></li><hr><li>Цитирование <div class=\\"markdown-help-container-example\\"> &gt;Text</div></li><hr></ul></div>"`;

exports[`workfield tests match snapshot: workfield snapshot 1`] = `
"<div class=\\"work-field\\"><div class=\\"workfield-actions\\"><div class=\\"workfield-content-edit-field-actions\\"><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"bold-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#bold-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"italic-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#italic-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"underscoped-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#underscoped-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"dotted-list-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#dotted-list-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"list-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#list-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"h1-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#h1-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"h2-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#h2-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"h3-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#h3-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"code-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#code-icon\\"></use></svg></div><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"table-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#table-icon\\"></use></svg></div><div aria-describedby=\\"popup-7\\"><div class=\\"small-button-container wf-action-btn\\" data-testid=\\"markup-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"15\\" height=\\"15\\"><use xlink:href=\\"test-file-stub#markup-icon\\"></use></svg></div></div></div><div class=\\"workfield-actions-container\\"><div class=\\"small-button-container\\" title=\\"Удалить заметку\\" data-testid=\\"remove\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"25px\\" height=\\"25\\"><use xlink:href=\\"test-file-stub#remove\\"></use></svg></div><div class=\\"small-button-container\\" title=\\"Скопировать заметку\\" data-testid=\\"copy\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"25px\\" height=\\"25\\"><use xlink:href=\\"test-file-stub#copy\\"></use></svg></div><div><div aria-describedby=\\"popup-8\\"><div class=\\"small-button-container\\" title=\\"Задать тэг\\" data-testid=\\"tag\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"25px\\" height=\\"25\\"><use xlink:href=\\"test-file-stub#tag\\"></use></svg></div></div></div><div><div aria-describedby=\\"popup-9\\"><div class=\\"small-button-container\\" title=\\"Поделиться\\" data-testid=\\"share\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"25px\\" height=\\"25\\"><use xlink:href=\\"test-file-stub#share\\"></use></svg></div></div></div><div class=\\"small-button-container\\" title=\\"Режим чтения\\" data-testid=\\"read-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"25px\\" height=\\"25\\"><use xlink:href=\\"test-file-stub#read-icon\\"></use></svg></div><div class=\\"medium-container\\" data-testid=\\"save-icon\\"><svg class=\\"image small-button-icon tooltip\\" fill=\\"BlackCommon\\" width=\\"25px\\" height=\\"25\\"><use xlink:href=\\"test-file-stub#save-icon\\"></use></svg><span>Сохранить</span></div></div></div><div class=\\"workfield-content\\"><div class=\\"workfield-content-edit-field\\"><div class=\\"workfield-content-edit-field-textarea\\"><textarea data-testid=\\"input-field\\">test-content</textarea></div></div><div data-testid=\\"pretty-view\\" class=\\"workfield-content-display-field\\"><p>test-content</p>
</div></div></div>"
`;
